src/main/java/com/exasol/udfdebugging/modules/udflogs/LogRecorder.java

```diff
@@ -0,0 +1,107 @@
+ */
+public final class LogRecorder implements AutoCloseable {
+    private static final Path LOG_DIRECTORY = Path.of("target/udf-logs");
+    private final Server server;
+
+    /**
+     * Create a new instance of {@link LogRecorder}.
+     * 
+     * @param logFileHandler callback for created log files
+     */
+    public LogRecorder(final Consumer<Path> logFileHandler) {
+        try {
+            if (!Files.exists(LOG_DIRECTORY)) {
+                Files.createDirectory(LOG_DIRECTORY);
+            }
+            this.server = new Server(logFileHandler);
+            new Thread(this.server).start();
+        } catch (final IOException exception) {
+            throw new UncheckedIOException(
+                    ExaError.messageBuilder("E-UDJ-18")
+                            .message("Failed to start server for retrieving UDF logs.", exception).toString(),
+                    exception);
+        }
+    }
+
+    @Override
+    public void close() throws Exception {
+        this.server.close();
+    }
+
+    /**
+     * Get the port the log recorder listens on.
+     * 
+     * @return port number
+     */
+    public int getPort() {
+        return this.server.getPort();
+    }
+
+    private static class Server implements Runnable, Closeable {
+        private final ServerSocket serverSocket;
+        private final Consumer<Path> logFileHandler;
+        private boolean running = true;
+
+        public Server(final Consumer<Path> logFileHandler) throws IOException {
+            this.logFileHandler = logFileHandler;
+            this.serverSocket = new ServerSocket(0);
+        }
+
+        int getPort() {
+            return this.serverSocket.getLocalPort();
+        }
+
+        @Override
+        public void run() {
+            try {
+                while (this.running) {
+                    final Socket client = this.serverSocket.accept();
+                    new Thread(new Logger(client, this.logFileHandler)).start();
+                }
+            } catch (final IOException exception) {
+                // ignore
+            }
+        }
+
+        @Override
+        public void close() throws IOException {
+            this.serverSocket.close();
+            this.running = false;
+        }
+    }
+
+    private static class Logger implements Runnable {
+        private final Socket socket;
+        private final Consumer<Path> logFileHandler;
+
+        private Logger(final Socket socket, final Consumer<Path> logFileHandler) {
+            this.socket = socket;
+            this.logFileHandler = logFileHandler;
+        }
+
+        @Override
+        public void run() {
+            try {
+                final Path logFile = Files.createTempFile(LOG_DIRECTORY, "udf-log-", ".txt");
+                this.logFileHandler.accept(logFile);
+                Files.copy(this.socket.getInputStream(), logFile, StandardCopyOption.REPLACE_EXISTING);
+            } catch (final IOException exception) {
```

Comment:
Here we should close the client socket.

In general, it would be good to add close method to Logger to close client socket (that should also close its inputstream).